editor/src/clj/editor/app_manifest.clj

```diff
@@ -0,0 +1,478 @@
+;; Copyright 2020-2022 The Defold Foundation
+;; Copyright 2014-2020 King
+;; Copyright 2009-2014 Ragnar Svensson, Christian Murray
+;; Licensed under the Defold License version 1.0 (the "License"); you may not use
+;; this file except in compliance with the License.
+;;
+;; You may obtain a copy of the License, together with FAQs at
+;; https://www.defold.com/license
+;;
+;; Unless required by applicable law or agreed to in writing, software distributed
+;; under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+;; CONDITIONS OF ANY KIND, either express or implied. See the License for the
+;; specific language governing permissions and limitations under the License.
+(ns editor.app-manifest
+  (:require [dynamo.graph :as g]
+            [editor.code.resource :as r]
+            [editor.code.data :as data]
+            [editor.resource-io :as resource-io]
+            [editor.graph-util :as gu]
+            [editor.code.util :as util]
+            [editor.yaml :as yaml]
+            [editor.properties :as properties]))
+
+(def windows #{:x86-win32 :x86_64-win32})
+
+(def android #{:armv7-android :arm64-android})
+
+(def linux #{:x86_64-linux})
+
+(def vulkan
+  #{:x86_64-osx
+    :x86_64-linux
+    :x86-win32 :x86_64-win32
+    :armv7-android :arm64-android
+    :arm64-ios})
+
+(def all-platforms
+  #{;; ios
+    :armv7-ios :arm64-ios :x86_64-ios
+    ;; android
+    :armv7-android :arm64-android
+    ;; osx
+    :x86_64-osx
+    ;; linux
+    :x86_64-linux
+    ;; windows
+    :x86-win32 :x86_64-win32
+    ;; web
+    :js-web :wasm-web})
+
+(def custom-lib-names
+  {:x86-win32 {"vpx" "vpx"
+               "vulkan" "vulkan-1"}
+   :x86_64-win32 {"vpx" "vpx"
+                  "vulkan" "vulkan-1"}})
+
+(defn platformify-excluded-lib [platform lib]
+  (or (-> custom-lib-names platform (get lib))
+      (and (contains? windows platform) (str "lib" lib))
+      lib))
+
+(defn platformify-lib [platform lib]
+  (or (some-> custom-lib-names platform (get lib) (str ".lib"))
+      (and (contains? windows platform) (str "lib" lib ".lib"))
+      lib))
+
+;; region toggles
+
+(defn contains-toggle [platform key value]
+  {:toggle :contains :platform platform :key key :value value})
+
+(defn boolean-toggle [platform key value]
+  {:toggle :boolean :platform platform :key key :value value})
+
+(defn exclude-libs-toggles [platforms libs]
+  (for [p platforms
+        l libs]
+    (contains-toggle p :excludeLibs (platformify-excluded-lib p l))))
+
+(defn libs-toggles [platforms libs]
+  (for [p platforms
+        l libs]
+    (contains-toggle p :libs (platformify-lib p l))))
+
+(defn generic-contains-toggles [platforms key values]
+  (for [p platforms
+        v values]
+    (contains-toggle p key v)))
+
+(defmacro ^{:arglists '([m & k+pred* default?])} get-in-guarded
```

Comment:
This is documentation. Actual args are different, but not very useful for the user (`[m & args]`). But when you use the documentation popup in IDEA it will show args from `:arglists` var meta if it's defined. Other IDEs should do the same, and it's used in `clojure.core`.